3DCalendar/CalendarView.swift

```diff
@@ -0,0 +1,166 @@
+        //return day == 7 ? 1 : day
+        return day
+    }
+    
+    func setUpViews() {
+        addSubview(myCollectionView)
+        myCollectionView.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        myCollectionView.leftAnchor.constraint(equalTo: leftAnchor).isActive = true
+        myCollectionView.rightAnchor.constraint(equalTo: rightAnchor).isActive = true
+        myCollectionView.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return numDaysPerMonth[currentMonthIndex-1] + firstWeekDayOfMonth - 1
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath) as! dateCell
+        cell.backgroundColor=UIColor.clear
+        if indexPath.item <= firstWeekDayOfMonth - 2 {
+            cell.isHidden = true
+        } else {
+            let calcDate = indexPath.row-firstWeekDayOfMonth+2
+            cell.isHidden = false
+            cell.label.text="\(calcDate)"
+            if calcDate < todaysDate && currentYear == presentYear && currentMonthIndex == presentMonthIndex {
+                cell.isUserInteractionEnabled=false
+                cell.label.textColor = UIColor.lightGray
+            } else {
+                cell.isUserInteractionEnabled=true
+                cell.label.textColor = UIColor.black
+            }
+        }
+        return cell
+    }
+
+}
+
+class dateCell: UICollectionViewCell {
+    
+    let label: UILabel = {
+        let lbl = UILabel()
+        lbl.text = "00"
+        lbl.textAlignment = .center
+        lbl.font = UIFont.systemFont(ofSize: 16)
+        lbl.textColor = UIColor.gray
+        lbl.translatesAutoresizingMaskIntoConstraints = false
+        return lbl
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        backgroundColor = UIColor.clear
+        layer.cornerRadius = 5
+        layer.masksToBounds = true
+        
+        setUpViews()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func setUpViews() {
+        addSubview(label)
+        label.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        label.leftAnchor.constraint(equalTo: leftAnchor).isActive = true
+        label.rightAnchor.constraint(equalTo: rightAnchor).isActive = true
+        label.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+    }
+}
+
+extension Date {
+    var weekday: Int {
+        return Calendar.current.component(.weekday, from: self)
+    }
+    var firstDayOfTheMonth: Date {
+        return Calendar.current.date(from: Calendar.current.dateComponents([.year,.month], from: self))!
+    }
+}
+
+extension String {
```

Comment:
Probably don't want this to live on `String` we can just make them live on the `DateFormatter` class itself